### PR TITLE
Ensure smooth audit-db syncs for h2, postgres, and mysql

### LINKED ISSUE #30887: BE - Make syncing with Audit DB work with all 3 app db types
It's currently failing for `h2`, and needs to be checked for `mysql`.

### PR DESCRIPTION
**Pull request by escherize**
resolves #30887

For audit-db, we pass in the app-db's `javax.sql.DataSource`, and when it's an h2 app-db, it won't have a connection string. If the db details are lacking a connetion string we shouldn't try to update it. This was causing a failure on h2 audit-db syncs.

---

- fixes typo: default-admin-db-id -> default-audit-db-id

- print db id without locale settings in `name-for-logging`
  - we probably never had a database with and id > 1000, but it would print like `1,000` in logs, when we'd prefer `1000`.

- remove 2 more `defendpoint-schema` forms

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30885)

### CODE REVIEW

File: src/metabase/driver/h2.clj [hunk 1]
```diff
@@ -467,7 +467,8 @@
 (defmethod sql-jdbc.conn/connection-details->spec :h2
   [_ details]
   {:pre [(map? details)]}
-  (mdb.spec/spec :h2 (update details :db connection-string-set-safe-options)))
+  (mdb.spec/spec :h2 (cond-> details
+                       (string? (:db details)) (update :db connection-string-set-safe-options))))
```
Comment (escherize) [inline src/metabase/driver/h2.clj:471]: Actual fix here. When we create the spec from the *application-db* we don't get a connection-string or db details. So the setting-conn-string-options step isn't needed.

File: src/metabase/sync/util.clj [hunk 2]
```diff
@@ -304,7 +304,7 @@
 
 (defmethod name-for-logging Database
   [{database-name :name, id :id, engine :engine,}]
-  (trs "{0} Database {1} ''{2}''" (name engine) (or id "") database-name))
+  (trs "{0} Database {1} ''{2}''" (name engine) (str (or id "")) database-name))
```
Comment (escherize) [inline src/metabase/sync/util.clj:307]: when `id` is a number, trs will apply locale settings. Therefore id `13371337` -> "13,371,337"; running it through `str` skips that.

**Review by noahmoss — APPROVED** [review]

File: enterprise/backend/src/metabase_enterprise/audit_db.clj [hunk 3]
```diff
@@ -16,19 +16,22 @@
...
-                           :id               default-admin-db-id
-                           :name             "Audit Database"
-                           :description      "Internal Audit DB used to power metabase analytics."
-                           :engine           engine
-                           :is_full_sync     true
-                           :is_on_demand     false
-                           :creator_id       nil
-                           :auto_run_queries true}))))
+     (do
+       ;; guard against someone manually deletihng the audit-db entry, but not removing the audit-db permissions.
```
[the remark this answers is not in the document]
Comment (escherize) [reply enterprise/backend/src/metabase_enterprise/audit_db.clj:24]: Good catch -- I'll fix this in the next PR

[the remark this answers is not in the document]
Comment (escherize) [reply enterprise/backend/src/metabase_enterprise/audit_db.clj:24]: https://github.com/metabase/metabase/pull/30994/commits/c62980184cd734db2179c4afc4cdf0f69aa81665#diff-fde45cb3759b54433232d76a54821b8b3105bb75ac7da16d3d8f8a9434081469R24

**Pull request merged**